code/controllers/ObjectCreatorPage_FrontEndWorkflowController.php

```diff
@@ -0,0 +1,256 @@
+	private static $allowed_actions = array(
+		'index',
+		'Form',
+		'doEdit',
+		'doFrontEndAction',
+	);
+
+	public $parentController = null;
+
+	public function __construct() {
+		$this->parentController = Controller::curr();
+		parent::__construct();
+		$this->request = $this->parentController->request;
+	}
+
+	public function index($request) {
+		if (!Member::currentUser()) {
+			 return $this->httpError(404);
+		}
+
+		$id = $request->param('Action');
+		if ($id == null)
+		{
+			return $this->parentController->customise(array(
+				'Title' => 'Review submissions',
+				'CreateForm' => '',
+				'Form' => '',
+				'IsReviewListing' => true,
+			));
+		}
+		else if ($id)
+		{
+			$workflowForm = $this->Form();
+			if (!$workflowForm)
+			{
+				return $this->httpError(404);
+			}
+			$page = $this->getContextObject();
+			$title = $page->Title;
+	        return $this->parentController->customise(array(
+	        	'Title' => 'Reviewing "'.$title.'" submission',
+	        	'CreateForm' => $workflowForm,
+	        	'Form' => $workflowForm,
+	        ));
+	    }
+	    return $this->httpError(404);
+	}
+
+	public function updateFrontendCreateForm($form) {
+		//
+		// Add 'Go to Review' button 
+		//
+		$fields = $form->Fields();
+		$idField = $fields->dataFieldByName('ID');
+		if ($idField 
+			&& $this->parentController->CreateType 
+			&& ($id = $idField->Value())) 
+		{
+			// Get Context Object and detect whether we can review/edit with workflows or not
+			$contextObj = $this->getContextObjectWithIDAndClass($id, $this->parentController->CreateType);
+			if ($contextObj)
+			{
+				$workflow = $contextObj->getWorkflowInstance();
+				if ($workflow && $workflow->CurrentAction()->canEditTarget($contextObj))
+				{
+					$actions = $form->Actions();
+					$actions->unshift(FormAction::create('doReview', 'Go to Review'));
+				}
+			}
+		}
+	}
+
+	public function updateFrontEndWorkflowActions($fields) {
+		$firstField  = $fields->first();
+		if (!$firstField instanceof FormAction) {
+			// Workaround for this bug on 3.3.1 or lower: https://github.com/silverstripe-australia/advancedworkflow/pull/242
+			return;
+		}
+
+		if ($this->parentController && !$this->parentController->AllowEditing) {
+			// Don't add edit button if can't edit
+			return;
+		}
+
+		$contextObj = $this->getContextObject();
+		if (!$contextObj)
+		{
+			return;
+		}
+		$workflow = $contextObj->getWorkflowInstance();
+		if (!$workflow->CurrentAction()->canEditTarget($contextObj))
+		{
+			return;
+		}
+
+		$fields->unshift(FormAction::create('doEdit', 'Edit'));
+	}
+
+	public function Form($request = null) {
+		$isExecutingAsAction = ($request && $request instanceof SS_HTTPRequest && $request->param('Action') === __FUNCTION__);
+		
+		$contextObj = $this->getContextObject();
+		if (!$contextObj)
+		{
+			if ($isExecutingAsAction) 
+			{
+				user_error('Cannot determine "getContextObject".', E_USER_WARNING);
```

Comment:
I want the errors to be shown during developer mode / only logged, not completely stop code execution.

https://docs.silverstripe.org/en/3.3/developer_guides/debugging/error_handling/

Since a lot of this functionality requires a variety of certain things to enabled/ configured in a particular way, I want to know exactly what I haven't configured or what broke, not get the code to silently fail.